rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/terminal/ITerminalIDEServer.java

```diff
@@ -470,12 +471,63 @@ public String getMainModule() {
             return mainModule;
         }
 
+        public @MonotonicNonNull ParserSpecification getPrecompiledParser() {
+            return precompiledParser;
+        }
+
         @Override
         public String toString() {
             return "LanguageParameter(pathConfig=" + pathConfig + ", name=" + name + ", extension=" + extension
-                + ", mainModule=" + mainModule + ", mainFunction=" + mainFunction + ")";
+                + ", mainModule=" + mainModule + ", mainFunction=" + mainFunction + ", precompiledParser=" + precompiledParser + ")";
+        }
+    }
+
+    public static class ParserSpecification {
+        /** absolute path to the file containing the precompiled parsers */
+        private final String parserLocation;
+        /** terminal to use from the defined parsers */
+        private final String nonTerminalName;
+        /** is the terminal a `start` terminal, default: true */
+        private final @Nullable Boolean nonTerminalIsStart;
+        /** allowAmbiguity (default is false) */
+        private final @Nullable Boolean allowAmbiguity;
+
+
+        public ParserSpecification(String parserLocation, String nonTerminalName, @Nullable Boolean nonTerminalIsStart, @Nullable Boolean allowAmbiguity) {
+            this.parserLocation = parserLocation;
+            this.nonTerminalName = nonTerminalName;
+            this.nonTerminalIsStart = nonTerminalIsStart;
+            this.allowAmbiguity = allowAmbiguity;
+        }
+
+        public ISourceLocation getParserLocation() throws FactTypeUseException {
```

Comment:
we have utilities for this elsewhere in the current package. looks like a clone.

Reply:
ouch, in the same exact file... I should have checked for that.